### PR TITLE
Improve request for placement domain event description

### PR DESCRIPTION
**Pull request by davidatkinsuk**
This description now include the dates.

Screenshot 2024-02-23 at 11 21 18

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1PlacementApplicationDomainEventService.kt [hunk 1]
```diff
@@ -42,7 +42,13 @@ class Cas1PlacementApplicationDomainEventService(
       deliusEventNumber = application.eventNumber,
       withdrawnAt = eventOccurredAt,
       withdrawnBy = domainEventTransformer.toWithdrawnBy(user),
-      withdrawalReason = placementApplication.withdrawalReason!!.name
+      withdrawalReason = placementApplication.withdrawalReason!!.name,
+      placementDates = placementApplication.placementDates.map {
```
Comment (pezholio) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1PlacementApplicationDomainEventService.kt:44]: Are we going to need to backfill these? Or are we not really concerned?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/domainevents/DomainEventDescriber.kt [hunk 2]
```diff
@@ -71,6 +75,15 @@ class DomainEventDescriber(
     return event.describe { "The assessment was appealed and ${it.eventDetails.decision.value}. The reason was: ${it.eventDetails.decisionDetail}" }
   }
 
+  private fun buildPlacementApplicationWithdrawnDescription(domainEventSummary: DomainEventSummary): String? {
```
Comment (pezholio) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/domainevents/DomainEventDescriber.kt:78]: It'd be nice if we could make the list more human with an `and` at the end of the last item, but I don't think it's a big issue

**Review by pezholio — APPROVED** [review]
Couple of comments, but LGTM on the whole

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1PlacementApplicationDomainEventService.kt (hunk 1 above)
Comment (davidatkinsuk) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1PlacementApplicationDomainEventService.kt:44]: These domain events have only been raised in pre-prod so happy to not bother in this case

**Pull request merged**